pulp_container/app/viewsets.py

```diff
@@ -511,6 +511,25 @@ class ContainerPullThroughRemoteViewSet(RemoteViewSet, RolesMixin):
         ],
     }
 
+    @extend_schema(
+        description="Trigger an asynchronous delete task",
+        responses={202: AsyncOperationResponseSerializer},
+    )
+    def destroy(self, request, pk, **kwargs):
+        """
+        Delete a model instance
+        """
+        # check if there is no distribution associated with this remote before removing it
+        linked_distribution = models.ContainerPullThroughDistribution.objects.filter(remote_id=pk)
+        if linked_distribution.exists():
+            log.info(
```

Comment:
This is basically too late. When this gets logged we still destroy the thing, so the damage has been done.
What if instead we mark the FK-relation `delete.PROTECT` instead?
Then the user needs to unset the remote on the distribution first. And that should not be an accidental step.

Reply:
When I opened the issue I thought that it would be fine to block the remote removal, then, when I started to work on the PR, I changed my mind and thought that blocking the removal and "forcing" users to first remove the associated distribution would be kind of "annoying" for users who really knows what they are doing and wanted to delete it.


>  _What if instead we mark the FK-relation delete.PROTECT instead?_

hum... to do so we would need to change the `remote` attribute from pulpcore Distribution which would impact all the other plugins, right?
https://github.com/pulp/pulpcore/blob/1672b3f405170d0a45de9c4b8cc2b059a066a162/pulpcore/app/models/publication.py#L659

I tried to override it in pulp_container
https://github.com/pulp/pulp_container/blob/bb7e18c090f50e75387cae2fbbe6f3025f901ae1/pulp_container/app/models.py#L688
but it seems like django does not allow it:
https://docs.djangoproject.com/en/5.1/topics/db/models/#field-name-hiding-is-not-permitted

> _"In normal Python class inheritance, it is permissible for a child class to override any attribute from the parent class. In Django, this isn’t usually permitted for model fields. If a non-abstract model base class has a field called author, you can’t create another model field or define an attribute called author in any class that inherits from that base class."_

Reply:
> hum... to do so we would need to change the `remote` attribute from pulpcore Distribution which would impact all the other plugins, right?

That may not be the worst idea...